classes/notification/NotificationDAO.inc.php

```diff
@@ -234,25 +234,20 @@ public function build($contextId, $level, $type, $assocType, $assocId, $userId =
      * @param int $notificationId
      * @param int $userId
      *
-     * @return bool
      */
     public function deleteById($notificationId, $userId = null)
     {
-        $params = [(int) $notificationId];
-        if (isset($userId)) {
-            $params[] = (int) $userId;
-        }
-        if ($this->update(
-            'DELETE FROM notifications WHERE notification_id = ?' . (isset($userId) ? ' AND user_id = ?' : ''),
-            $params
-        )) {
-            // If a notification was deleted (possibly validating
-            // $userId in the process) delete associated settings.
-            $notificationSettingsDao = DAORegistry::getDAO('NotificationSettingsDAO'); /** @var NotificationSettingsDAO $notificationSettingsDaoDao */
-            $notificationSettingsDao->deleteSettingsByNotificationId($notificationId);
-            return true;
+        $notificationSettingsDao = DAORegistry::getDAO('NotificationSettingsDAO'); /** @var NotificationSettingsDAO $notificationSettingsDaoDao */
+        $notificationSettingsDao->deleteSettingsByNotificationId($notificationId);
+
+        $query = DB::table('notifications')
```

Comment:
Here and elsewhere -- not a big deal (no need to change it here), but rather than...
```php
$query = DB::table('someTable');

$query->someOperation();
if ($condition) $query->someOtherOperation();

$result = $query->execute();
```

...I prefer the more readable fluent formulation...

```php
$result = DB::table('someTable')
    ->someOperation()
    ->when($condition, function($query) {
        $query->someOtherOperation();
    })
    ->execute();
```

Reply:
I will keep that in mind, thanks